src/oc_erchef/apps/chef_objects/test/chef_s3_tests.erl

```diff
@@ -75,19 +101,38 @@ setup_s3(InternalS3Url, ExternalS3Url) ->
     meck:expect(mini_s3, new, 3, mock_mini_s3_ctx),
     meck:expect(mini_s3, get_object_metadata, 4, mock_metadata).
 
+choose_url_style(InternalS3Url, ExternalS3Url) ->
+    case {InternalS3Url, ExternalS3Url} of
+        {_, host_header} ->
+            fun get_host_port/1;
+%        {Same, Same} ->
+%            fun get_host_noport/1;
+        _ ->
+            fun get_host_noport/1
+    end.
+
 generate_presigned_url_uses_configured_s3_url_test_() ->
     MockedModules = [mini_s3, chef_secrets_mock_provider],
     HostHeaderUrl = "https://api.example.com:443",
     OrgId = <<"deadbeefdeadbeefdeadbeefdeadbeef">>,
     Checksum = <<"aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa">>,
     Lifetime = 3600,
-    Expect_s3_url = fun(ExpectMethod, ExpectUrl, ExpectLifetime) ->
+%    Config = mini_s3:new("", "", "http://s3.amazonaws.com"),
+    Expect_s3_url = fun(ExpectMethod, ExpectUrl, _ExpectLifetime) ->
                             meck:expect(mini_s3, s3_url,
-                                        fun(HTTPMethod, Bucket, _Key, MyLifetime, _ContentMD5,
-                                            #config{s3_url = S3Url}) ->
+                                        fun(HTTPMethod, Bucket, _Key, _MyLifetime, _ContentMD5,
+                                            Config) ->
                                                 ?assertEqual(ExpectMethod, HTTPMethod),
                                                 ?assertEqual("testbucket", Bucket),
-                                                ?assertEqual(ExpectLifetime, MyLifetime),
+                                                % CODE REVIEW: disable expiry window. expiry windows were redone,
+                                                % and are tested elsewhere
+%                                               ?assertEqual(ExpectLifetime, MyLifetime),
```

Comment:
It might be good to check this as a part of this as well. (If it is not too hard).

Could you add a pointed to where this is tested?

Reply:
Tested at: `mini_s3_tests:make_expire_win_test`

Reply:
Added pointer.